vault/mount_test.go

```diff
@@ -808,6 +829,23 @@ func testCore_MountTable_UpgradeToTyped_Common(
 		t.Fatalf("bad: values here should be different")
 	}
 
+	// Remove any transactional storage entries: we want to replace the mount
+	// table with a pre-transactional variant to force upgrades to be run.
+	if _, ok := c.barrier.(logical.TransactionalStorage); ok && testType != "audits" {
+		postTxnEntries, err := c.barrier.List(context.Background(), path+"/")
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, txnEntry := range postTxnEntries {
+			t.Logf("removing entry: %v", path+"/"+txnEntry)
+			if err := c.barrier.Delete(context.Background(), path+"/"+txnEntry); err != nil {
+				t.Fatal(err)
+			}
+		}
+	}
+
+	// Write the pre-typed version
```

Comment:
@JanMa @DanGhita This existing test is sufficient to handle the migration case, IMO, from a single-entry table to a multi-entry table.